### ISSUE TITLE
Add comparison against kd-tree

### ISSUE THREAD

**Issue by RReverser** [cavemanloverboy/bosque#7]
The somewhat older [kd_tree](https://crates.io/crates/kd_tree) crate has mentioned optimisations as well (it uses quickselect, as well as has parallel build via par_build_by_ordered_float) so it could be interesting to compare its performance to this implementation, but I noticed it's currently missing in README.

(Originally posted at https://github.com/cavemanloverboy/FNNTW/issues/8)

**Comment by clbarnes** [comment]
Kiddo is another one we've found to be very fast! https://github.com/sdd/kiddo But I imagine it depends quite a lot on CPU cache and point cloud size as bosque optimises heavily for that

**Comment by RReverser** [comment]
Maybe. I've been using a personal fork of kd-tree for quite a while with a few extra cache-friendly optimisations that I haven't yet had time to upstream, but even the upstream version should be pretty fast, as it uses a plain old flat `Vec<T>` for its tree.

That's why I'm curious whenever I see a new kd-tree library in Rust to see if it can beat that flat & parallelised representation.

**Comment by clbarnes** [comment]
The best way to approach this would probably be to add bosque and kd_tree to this project: https://github.com/sdd/kd-tree-comparison

**Comment by RReverser** [comment]
Well that's a separate project; the README of this project has its own benchmarks, so for now I'm just interested in seeing kd-tree here.

**Comment by clbarnes** [comment]
Right, but for `N` spatial lookup libraries, there are `N * (N - 1)` possible benchmarks if all libraries must be represented in all libraries' benchmarks, and every new library must implement `N` benchmarks, and every library maintainer needs to track every new entrant into the space, so it's much harder generally. If we, as a community, can converge on a single spatial lookup benchmark suite, then each new library only has to implement 1, and it's much easier to find the state of the art performance for people who just want to use a lookup library with minimum fuss.

**Comment by RReverser** [comment]
> If we, as a community, can converge on a single spatial lookup benchmark suite

Mandatory https://xkcd.com/927/ reference :) That never works that way.

Look, I just opened an issue on a library I'm interested in. If you want to add benchmarks to that other project you mentioned, please go for it, but that project and the discussion is a bit out of scope of _my_ issue.
